matchzoo/preprocessor/process_units.py

```diff
@@ -170,24 +170,27 @@ def transform(self, tokens: list) -> list:
         return [lemmatizer.lemmatize(token, pos='v') for token in tokens]
 
 
-class NgramLetterUnit(StatefulProcessorUnit):
+class NgramLetterUnit(ProcessorUnit):
     """
     Process unit for n-letter generation.
 
     Triletter is used in :class:`DSSMModel`.
-    This processor is expected to execute after `Vocab`
+    This processor is expected to execute before `Vocab`
     has been created.
 
-    Returned `input_dim` is the dimensionality of :class:`DSSMModel`.
-    """
+    Examples:
+        >>> triletter = NgramLetterUnit()
+        >>> rv = triletter.transform(['word'])
+        >>> len(rv)
+        4
+        >>> '#wo' in rv
+        True
```

Comment:
Try printing a sorted version of all tokens. Even better, move line 195:196 here to make the entire process clear.